src/platforms/c.rkt

```diff
@@ -0,0 +1,299 @@
+#lang racket
+
+;;; C platform:
+;;; C/C++ on Linux with a full libm
+
+(require math/bigfloat
+         math/flonum
+         "runtime/float32.rkt" ; float representation helper functions
+         "runtime/libm.rkt"    ; libm wrapper
+         "../utils/float.rkt"  ; for shift/unshift
+         "../syntax/platform.rkt")
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;; EMPTY PLATFORM ;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define 64bit-move-cost   0.12538399999999972)
+(define 32bit-move-cost   0.12961999999999974)
+(define boolean-move-cost 0.1)
+
+(define c-platform
+  (make-empty-platform 'c #:if-cost boolean-move-cost))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;; BOOLEAN ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;; representation ;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define bool
+  (make-representation #:name 'bool
+                       #:type 'bool
+                       #:repr? boolean?
+                       #:bf->repr identity
+                       #:repr->bf identity
+                       #:ordinal->repr (λ (x) (= x 0))
+                       #:repr->ordinal (λ (x) (if x 1 0))
+                       #:total-bits 1
+                       #:special-value? (const #f)))
+
+(platform-register-representation! c-platform #:repr bool #:cost boolean-move-cost)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;; constants ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(platform-register-implementations!
+ c-platform
+ ([TRUE  () bool (TRUE)  (const true)  (! TRUE)  boolean-move-cost]
+  [FALSE () bool (FALSE) (const false) (! FALSE) boolean-move-cost]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;; operators ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define (and-fn . as)
+  (andmap identity as))
+(define (or-fn . as)
+  (ormap identity as))
+
+(platform-register-implementations!
+ c-platform
+ ([not ([x : bool])            bool (not x)   not    (not x)   boolean-move-cost]
+  [and ([x : bool] [y : bool]) bool (and x y) and-fn (and x y) boolean-move-cost]
+  [or  ([x : bool] [y : bool]) bool (or x y)  or-fn  (or x y)  boolean-move-cost]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;; BINARY 32 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;; representation ;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define binary32
+  (make-representation #:name 'binary32
+                       #:type 'real
+                       #:repr? flonum?
+                       #:bf->repr bigfloat->float32
+                       #:repr->bf bf
+                       #:ordinal->repr (shift 31 ordinal->float32)
+                       #:repr->ordinal (unshift 31 float32->ordinal)
+                       #:total-bits 32
+                       #:special-value? nan?))
```

Comment:
I still think the representations should be shipped in the main Herbie libraries, not in a platform. All three representations: boolean, binary32, and binary64.